knpu/episode3/pdo-ship-storage.md

```diff
@@ -28,46 +33,81 @@ This leaves us needing to make our `ShipLoader` more generic.
 Right now, all of the logic of querying things from the database is hardcoded in here. 
 So let's create a new class whose only job is to load ship data through the database, or PDO. 
 
-Create a new class called `PdoShipStorage`. Looking back inside `ShipLoader` there are two
-types of queries that we make. Sometimes we query for all of the ships and sometimes we query
-for just one ship by ID. 
+Create a new class called `PdoShipStorage`:
+
+[[[ code('c5b654f23f') ]]]
+
+Looking back inside `ShipLoader` there are two types of queries that we make:
+
+[[[ code('a1d793a973') ]]]
+
+Sometimes we query for all of the ships and sometimes we query for just one ship by ID. 
 
 Back to our `PdoShipStorage` I'll create two methods, to cover both of those actions. First,
 create a `public function fetchAllShipsData()` which we'll fill out in just one second. Now,
-add `public function fetchSingleShipData()` and pass it the id that we want to query for. 
+add `public function fetchSingleShipData()` and pass it the id that we want to query for:
+
+[[[ code('3cd0c8eb48') ]]]
+
+Before we go any further head back to our `bootstrap.php` file and make sure that we require this:
+
+[[[ code('f8879801b8') ]]]
 
-Before we go any further head back to our `boostrap.php` file and make sure that we require this. 
 Perfect!
 
 What I want to do is move all the querying logic from `ShipLoader` into this `PdoShipStorage` class.
-Let's start with the logic that queries for one ship and pasting that over here. Notice, that we're
-not returning an object here this is just a really dumb class that returns data, an array in our case.
+Let's start with the logic that queries for one ship and pasting that over here:
+
+[[[ code('11bedaf776') ]]]
```

Comment:
Hmm, I see we *do* reference the `$this->pdo` versus `$this->getPdo()` - I didn't see that before. I'll still leave it alone, but that one step should maybe have been split into 2 pieces so we could grab the separate steps that are happening here.